File: Web/src/api/groupChallenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetGroupChallengeList(data) {
    return apiClient.post("/groupChallenge/GetGroupChallengeList", data).then(handleResponse).catch(handleError)
}

function GetGroupChallenge(data) {
    return apiClient.post("/groupChallenge/GetGroupChallenge", data).then(handleResponse).catch(handleError)
}

function CreateGroupChallenge(data) {
    return apiClient.post("/groupChallenge/CreateGroupChallenge", data).then(handleResponse).catch(handleError)
}

function UpdateGroupChallenge(data) {
    return apiClient.post("/groupChallenge/UpdateGroupChallenge", data).then(handleResponse).catch(handleError)

}

function DeleteGroupChallenge(data) {
    return apiClient.post("/groupChallenge/DeleteGroupChallenge", data).then(handleResponse).catch(handleError)

}

const groupChallengeApi = {
  GetGroupChallengeList,
  GetGroupChallenge,
  CreateGroupChallenge,
  UpdateGroupChallenge,
  DeleteGroupChallenge
}

export default groupChallengeApi

File: Server/route/groupChallenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetGroupChallengeRoute(r *gin.Engine) {
	groupChallenge := r.Group("/groupChallenge")
	groupChallenge.Use(middleware.JWTAuthMiddleware())
	{
		groupChallenge.POST("/GetGroupChallengeList", api.GetGroupChallengeList)
        groupChallenge.POST("/GetGroupChallengeListByQuery", api.GetGroupChallengeListByQuery)
		groupChallenge.POST("/GetGroupChallenge", api.GetGroupChallenge)
	}
	adminGroupChallenge := r.Group("/groupChallenge")
	adminGroupChallenge.Use(middleware.AdminAuthMiddleware())
	{
		adminGroupChallenge.POST("/CreateGroupChallenge", api.CreateGroupChallenge)
		adminGroupChallenge.POST("/UpdateGroupChallenge", api.UpdateGroupChallenge)
		adminGroupChallenge.POST("/DeleteGroupChallenge", api.DeleteGroupChallenge)
	}
}
